### PR TITLE
PR: Do not check for change on every keystroke

### PR DESCRIPTION
**Pull request by impact27**
## Description of Changes
I am not sure why it is necessery to check for the change status on every text change. Upless I am missing something this is not needed.

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

### Issue(s) Resolved

Fixes #

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:

### CODE REVIEW

File: spyder/plugins/editor/widgets/base.py [hunk 1]
```diff
@@ -214,10 +213,6 @@ def clear_extra_selections(self, key):
         self.extra_selections_dict[key] = []
         self.update()
 
-    def changed(self):
-        """Emit changed signal"""
-        self.modificationChanged.emit(self.document().isModified())
```
[the remark this answers is not in the document]
Comment (impact27) [reply spyder/plugins/editor/widgets/base.py:219]: But this signal should be emitted automatically by Qt on state change? I tried it and I could see the state changing correctly without this. Unless I am missing something it should work without it.

[the remark this answers is not in the document]
Comment (impact27) [reply spyder/plugins/editor/widgets/base.py:219]: See https://doc.qt.io/qt-5/qtextdocument.html#modificationChanged

[the remark this answers is not in the document]
Comment (ccordoba12) [reply spyder/plugins/editor/widgets/base.py:219]: Ok, I didn't know that. It seems this functionality was added when Spyder was created, probably to account for it not being available at the time in Qt.

**Review by ccordoba12 — APPROVED** [review]
Looks good to me, thanks @impact27!

**Pull request merged**